pkg/sync/syncWorker.go

```diff
@@ -362,8 +365,19 @@ func (s *syncWorker) validateAndInsertEnvelope(
 		lastNs = stream.lastEnvelope.OriginatorNs()
 	}
 	if env.OriginatorSequenceID() != lastSequenceID+1 || env.OriginatorNs() < lastNs {
-		// TODO(rich) Submit misbehavior report and continue
-		s.log.Error("Received out of order envelope")
+		if report, err := misbehavior.NewSafetyFailureReport(
```

Comment:
I really struggle to read this inlined error handling. We generally don't use the format.

The explicit formatting is much more readable:
```
report, err := misbehavior.NewSafetyFailureReport(
    stream.nodeID,
    message_api.Misbehavior_MISBEHAVIOR_OUT_OF_ORDER,
    true,
    []*envUtils.OriginatorEnvelope{stream.lastEnvelope, env},
)
if err != nil {
    // Log the error if the report creation fails
    s.log.Debug("Failed to create misbehavior report", zap.Error(err))
    return // Exit early or handle it as appropriate
}

// Submit the misbehavior report
err = s.misbehaviorService.SafetyFailure(report)
if err != nil {
    // Log the error if the report submission fails
    s.log.Debug("Failed to submit misbehavior report", zap.Error(err))
}
```

Reply:
Fair enough. I don't love the double nesting.

I'll move to its own function so that I can early return, since this whole code path is optional